File: CSE3063F22_iteration2_classes/Course.java
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;


public class Course {

        private String courseId;
        private String courseName;
        private List<String> prerequisites;
        private int Seatlimit;
        private String Semester;

        public Course(String courseId) throws FileNotFoundException, IOException, ParseException {
            this.courseId = courseId;
            Object obj = new JSONParser().parse(new FileReader("courses/"+getCourseId()+".json"));
        JSONObject jo = (JSONObject) obj;

        String Cname = (String) jo.get("CourseName");
        setCourseName(Cname);

        JSONArray prereq = (JSONArray) jo.get("Prereq");
        this.prerequisites = new ArrayList<>();
        setPrerequisites(prereq);
        
        String seat_limit = (String) jo.get("Seatlimit");
        int Seat_lim=Integer.parseInt(seat_limit);
        setSeatLimit(Seat_lim);

       
        String CSemester = (String) jo.get("CourseType");
        setSemester(CSemester.substring(0, 1));
                
        }

        public String getCourseId() {
            return courseId;
        }

        // public void setCourseId(passedCourses) {
        //     this.courseId = passedCourses;
        // }

        public String getCourseName() {
            return courseName;
        }

        public void setCourseName(String courseName) {
            this.courseName = courseName;
        }

        public int getSeatLimit() {
            return Seatlimit;
        }

        public void setSeatLimit(int limit) {
            this.Seatlimit = limit;
        }
        
        public String getSemester() {
            return Semester;
        }

        public void setSemester(String semester) {
            this.Semester = semester;
        }
        public List<String> getPrerequisites() {
            return prerequisites;
        }

        public void setPrerequisites(List<String> prerequisites) {
            this.prerequisites = prerequisites;
        }
        
        public boolean isEnoughSeatLimit(){
            boolean isIt=true;
            int seat_limit = getSeatLimit();
            
    
            if(seat_limit<=0){
                isIt=false;
            }
    
            return isIt;
    
        }


        
         public void updateSeat_Limit(String courseid) throws FileNotFoundException, IOException, org.json.simple.parser.ParseException{
        System.out.println(courseid);
        Object obj = new JSONParser().parse(new FileReader("courses/"+courseid+".json"));
        JSONObject jo = (JSONObject) obj;
        int value = Integer.parseInt(String.valueOf(jo.get("Seatlimit")));
        String new_value = String.valueOf(value-1);
        
        jo.remove("Seatlimit");
        jo.put("Seatlimit", new_value);

        FileWriter writer = new FileWriter("courses/"+courseid+".json",false);
        writer.write(jo.toString());
        writer.close();
     

    }



    }

File: python_code/Course.py
import json
import os
import random

class Course:
    def __init__(self, course_id):
        self.__course_id = course_id
        try:
            with open(f'courses/{self.__course_id}.json') as f:
             data = json.load(f)
            self.__course_name = data['CourseName']
            self.__prerequisites = data['Prereq']
            self.__seat_limit = data['Seatlimit']
            self.__semester = data['CourseType'][0]
        except FileNotFoundError:
                print ("Not found course file!")
      
    
    
    def get_course_id(self):
        return self.__course_id
    
    """   def get_course_name(self):
        return self.course_name
 """
    
    """
    def set_course_name(self, course_name):
        self.course_name = course_name """
    
    def get_prerequisites(self):
        return self.__prerequisites
        
    
    
    """def set_prerequisites(self, prerequisites):
        self._prerequisites = prerequisites"""
    
    @property
    def seat_limit(self):
        return self.__seat_limit
    
    @seat_limit.setter
    def seat_limit(self, seat_limit):
        self.__seat_limit = seat_limit
    
    
    def getSemester(self):
        return self.__semester
    
   
    def setSemester(self, semester):
        self.__semester = semester
    
    def is_enough_seat_limit(self):
        return int(self.__seat_limit) > 0
    
    def update_seat_limit(self, course_id):
        try:
            with open(f'courses/{course_id}.json') as f:
                data = json.load(f)
            data['Seatlimit'] = str(int(data['Seatlimit'])-1)
            with open(f'courses/{course_id}.json', 'w') as f:
                json.dump(data, f)
            print("Seat limit of "+course_id+" is updated succesfully.")
        except FileNotFoundError:
            print ("not found file error occured")